docs/cow-protocol/concepts/flashloans/solvers.md

```diff
@@ -0,0 +1,51 @@
+---
+sidebar_position: 4
+---
+
+# Solvers
+
+The solver receives for each order an optional object indicating the flashloan's hints. This information serves as a hint for the solver, but the solver ultimately can modify this data in order to make the operation more optimal.
+
+## How to get the flashloan's hints
+
+The user is able to create a flashloan order's hint by attaching to the `appData` the specified metadata. The autopilot reads the order and cuts it into a [batch auction](../introduction/batch-auctions). Then the driver fetches the `appData` by calling the orderbook with `GET /v1/app_data/<app_data_hash>` for every order and caches them in memory. The driver includes the flashloan information into the batch auction's order before sending it to the solver(s).
```

Comment:
There are some sections of the docs which are specifically geared towards solvers or the driver. Instead of having this information as part of the `flashloans` section the driver and solver sections should each get a flashloan section.
Also how the driver fetches the appdata / flashloan hint is irrelevant for the solver. The solver mostly cares about the format the data is presented, how it should report flashloans in its solution and how this affects their optimization problem:
* at what point in time the funds will be available
* who will receive the funds
* gas overhead
* how they are supposed to pay the funds back (with reference to the behavior of the reference driver and how they could do it themselves if they have a custom driver).